godtools/App/Services/Renderer/ManifestResourcesCache/ManifestResourcesCache.swift

```diff
@@ -7,25 +7,26 @@
 //
 
 import UIKit
+import GodToolsToolParser
 
-class ManifestResourcesCache: ManifestResourcesCacheType {
```

Comment:
@levieggertcru could you remind me why you're removing the protocols?  I'm assuming it is because it is easier/cleaner to just reference the classes directly.  But it makes me curious, when is it preferable to use protocols and when is it preferable to not use them?  Should you only use them when you expect to have multiple classes that have the same properties?  Or when you anticipate switching out one service for another?  I'm curious what the best practices are when it comes to protocols.

Reply:
Hey @reldredge71 that's a good question.  I initially created those protocols when switching from the XML parser to GodToolsToolParser.  It was a big transition and I needed to be able to switch between both the XML parser and GodToolsTool parser.  I decided to remove the protocols for a couple of reasons. 
1. I don't see us needing to make GodToolsToolParser interchangeable just because we maintain it and Android, iOS, and Web are dependent on it.
2. It created a lot of extra files and overhead.  For every type in the GodToolsToolParser we'd need to create 2 additional files (the protocol and either an extension or concrete wrapper class to implement the protocol on GodToolsToolParser models).  This created additional overhead when parsing models to view / viewModel, not sure how much it impacted rendering performance, but it could impact it some.

Protocols are suited for abstraction.  See the Interface Segregation Principle ( https://reflectoring.io/interface-segregation-principle/#:~:text=The%20Interface%20Segregation%20Principle%20(ISP,%E2%80%9D%20or%20%E2%80%9Cfat%E2%80%9D%20interface )
They're also good to make things interchangeable.  In Clean Architecture they talk about using them to make the data layer interchangeable with the domain layer. 
They also act similarly as a header file in Objective-C.  We've been using them for our ViewModels which make it easy to see what our ViewModel exposes to the View.

Reply:
Also see Protocol Oriented Programming -> https://scotteg.github.io/protocol-oriented-programming

Reply:
Hey @reldredge71 also a good read https://docs.swift.org/swift-book/LanguageGuide/Protocols.html